airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/__init__.py

```diff
@@ -1,99 +1,43 @@
-    connector_registry_location_markdown,
-    connector_registry_location_html,
-)
-from orchestrator.assets.registry import (
-    oss_destinations_dataframe,
-    cloud_destinations_dataframe,
-    oss_sources_dataframe,
-    cloud_sources_dataframe,
-    oss_registry_from_metadata,
-    cloud_registry_from_metadata,
-    legacy_cloud_sources_dataframe,
-    legacy_oss_sources_dataframe,
-    legacy_cloud_destinations_dataframe,
-    legacy_oss_destinations_dataframe,
-    legacy_cloud_registry_dict,
-    legacy_cloud_registry,
-    legacy_oss_registry_dict,
-    legacy_oss_registry,
-)
-from orchestrator.assets.metadata import (
-    legacy_registry_derived_metadata_definitions,
-    valid_metadata_report_dataframe,
-    metadata_definitions,
-)
-
-from orchestrator.assets.dev import (
-    persist_metadata_definitions,
-    overrode_metadata_definitions,
-    oss_registry_diff,
-    cloud_registry_diff,
-    cloud_registry_diff_dataframe,
-    oss_registry_diff_dataframe,
-    metadata_directory_report,
-    oss_registry_diff_report,
-    cloud_registry_diff_report,
+from orchestrator.assets import (
+    github,
+    specs_secrets_mask,
+    spec_cache,
+    registry,
+    registry_report,
+    legacy_registry,
+    metadata,
+    dev,
 )
 
-from orchestrator.jobs.registry import generate_registry_markdown, generate_local_metadata_files, generate_registry
+from orchestrator.jobs.registry import generate_registry_reports, generate_local_metadata_files, generate_registry
 from orchestrator.sensors.registry import registry_updated_sensor
 from orchestrator.sensors.metadata import metadata_updated_sensor
 
 from orchestrator.config import REPORT_FOLDER, REGISTRIES_FOLDER, CONNECTORS_PATH, CONNECTOR_REPO_NAME
 from metadata_service.constants import METADATA_FILE_NAME, METADATA_FOLDER
 
-ASSETS = [
-    all_destinations_dataframe,
-    all_sources_dataframe,
-    all_specs_secrets,
-    cached_specs,
-    cloud_destinations_dataframe,
-    cloud_registry_diff_dataframe,
-    cloud_registry_diff_report,
-    cloud_registry_diff,
-    cloud_registry_from_metadata,
-    cloud_sources_dataframe,
-    connector_registry_location_html,
-    connector_registry_location_markdown,
-    github_connector_folders,
-    legacy_cloud_destinations_dataframe,
-    legacy_cloud_registry_dict,
-    legacy_cloud_registry,
-    legacy_cloud_sources_dataframe,
-    legacy_oss_destinations_dataframe,
-    legacy_oss_registry_dict,
-    legacy_oss_registry,
-    legacy_oss_sources_dataframe,
-    legacy_registry_derived_metadata_definitions,
-    metadata_definitions,
-    metadata_directory_report,
-    oss_destinations_dataframe,
-    oss_registry_diff_dataframe,
-    oss_registry_diff_report,
-    oss_registry_diff,
-    oss_registry_from_metadata,
-    oss_sources_dataframe,
-    overrode_metadata_definitions,
-    persist_metadata_definitions,
-    specs_secrets_mask_yaml,
-    valid_metadata_report_dataframe,
-]
+ASSETS = load_assets_from_modules(
```

Comment:
👍 . But according to Dagster simple example [here](https://docs.dagster.io/tutorial/writing-your-first-asset) they never mention the list of exposed assets should be explicitly declared. I looks mandatory in our codebase though, because I spent a fair amount of time don't understanding why my new assets was not available in the UI.

Reply:
Ah nice catch!

So this is a failure of how Dagster does documentation IMO

You expect the decorators to wire everything up.

And their docs are paged in a way that makes it look like that.

But if you go further back in their tutorial you will see they call a project scaffold function
https://docs.dagster.io/tutorial/setup

Which creates an `__init__.py` file with this template:
https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_generate/templates/CODE_LOCATION_NAME_PLACEHOLDER/CODE_LOCATION_NAME_PLACEHOLDER/__init__.py

Reply:
Thank you for clarifying!